shell/platform/tizen/channels/platform_channel.cc

```diff
@@ -24,6 +35,94 @@ PlatformChannel::PlatformChannel(flutter::BinaryMessenger* messenger)
 
 PlatformChannel::~PlatformChannel() {}
 
+#if defined(MOBILE) || defined(WEARABLE)
+namespace {
+
+class FeedbackManager {
+ public:
+  enum class ResultCode {
+    OK,
+    NOT_SUPPORTED_ERROR,
+    PERMISSION_DENIED_ERROR,
+    UNKNOWN_ERROR
+  };
+
+  FeedbackManager() {
+    FT_LOGD("Enter FeedbackManager::FeedbackManager()");
+
+    auto ret = feedback_initialize();
+    if (FEEDBACK_ERROR_NONE != ret) {
+      FT_LOGD("feedback_initialize() failed with error: [%d] (%s)", ret,
```

Comment:
Could you use ```FT_LOGE``` or ```FT_LOGW``` to log errors?

Reply:
I have thought about it, but am not sure, if I should use `FT_LOGE`/`FT_LOGW`. [Flutter's code](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#only-log-actionable-messages-to-the-console) style requires to only log 
"actual errors and actionable warnings (warnings that can always be dealt with and fixed".

Messages logged with `FT_LOGE` are shown in the output of `flutter-tizen run` so I have assumed, they are meant for Flutter application developers.
Error messages like the one in 139th line are only relevant for flutter engine developers, but not for app developers. I log all of them with `FT_LOGD`, so that they're hidden from app developers, who can do nothing about them and I only use `FT_LOGE`/`FT_LOGW` for error messages, that app devs can understand and take some steps to fix - e.g. a message about a missing privilege.

However, I can't see messages logged with `FT_LOGE`/`FT_LOGW` in Flutter's DevTools. Am I right that messages shown in `flutter-tizen run` output are meant for application developers? What do you think about my logging policy - is it right?

Reply:
@pwasowski2 I totally agree with the policy but in my humble opinion the initialization failure here looks like an "actual" error that should be logged. If you search for use cases of `FML_LOG(ERROR)` in other platform implementations (`shell/platform`), there are still many places where error messages are logged out even though the user can't do anything about them.

The Flutter engine has a switch for choosing log verbosity which can be activated by providing the `--verbose-system-logs` option to the `flutter run` command. The min log level defaults to LOG_ERROR as you can see in:
https://github.com/flutter-tizen/engine/blob/9dfe48b069c2e092e848466c16ddc94db37cdc43/shell/common/shell.cc#L229-L230 The Tizen embedder doesn't support this option yet (we need to overhaul our logging facility and clean up all the messy logs first). However you will see some `FML_LOG(INFO)` and `FML_LOG(WARNING)` messages in other platform implementations - those are not logged to the user console by default.

Reply:
@swift-kim, thanks for your answer :)

I have replaced `FT_LOGD` with `FT_LOGE` for error messages.

What do you think, about adding more logging functions/macros, that we could use to log messages about internal errors to dlog without logging them to Flutter console?

For example:

```c
FT_LOG_E_DBG("feedback_initialize() failed with error: [%d] (%s)", ret, get_error_message(ret));
FT_LOG_W_DBG("feedback_initialize() failed with error: [%d] (%s)", ret, get_error_message(ret));
```

Reply:
@pwasowski2 The `--verbose-system-logs` option is meant for such internal log filtering. Is there any particular goal in your mind? The simplest way to implement `--verbose-system-logs` for Tizen is to read the `fml::GetMinLogLevel()` value and set some filters on `FT_LOGD`/`FT_LOGI`/`FT_LOGW` based on it in `tizen_log.h`.

Reply:
@swift-kim Thank you for your kind explanation, maybe... I should have left a more detailed comment from the beginning.
@pwasowski2 Thank you for your additional works

Reply:
@swift-kim, I have meant logging functions/macros, that would allow to log messages about details of internal engine errors (e.g. error codes returned from native functions) without showing these messages to the app developers - something like:

| Function | Message visible in Flutter Console | Message visible in Tizen's dlog |
| --------------- | --------------- | --------------- |
| FT_LOGE | ✔ | ✔ |
| FT_LOGE_DBG | ❌  | ✔ |
| FT_LOGE | ✔ | ✔ |
| FT_LOGE_DBG | ❌ | ✔ |

Messages logged with `FT_LOGE_DBG`/`FT_LOGW_DBG` would be helpful for investigation of problems, reported by users of release Tizen builds, which don't log messages with `D` severity to dlog.

Reply:
@pwasowski2 Well, if we implement `--verbose-system-logs`, it becomes like this:

| Severity | `--verbose-system-logs` is off (default) | `--verbose-system-logs` is on |
|:-:|:-:|:-:|
| E | Visible in Flutter Console<br>Visible in Tizen's dlog | Visible in Flutter Console<br>Visible in Tizen's dlog |
| W | **Invisible** | Visible in Flutter Console<br>Visible in Tizen's dlog |
| I | **Invisible** | Visible in Flutter Console<br>Visible in Tizen's dlog |
| D | **Invisible** | **Invisible** in Flutter Console<br>Visible in Tizen's dlog |

Note that eventually we have to reduce the number of `FT_LOGD` messages (or make more meaningful) and change them all to `FT_LOGI`. After that, the user can simply use `--verbose-system-logs` to investigate/report problems in more detail (otherwise they will see only `FT_LOGE` messages by default).

Also it's the engine developer's responsibility to extract meaningful information from the error code returned by native functions and show to the user. Are you thinking of logging a "partial" error message (summary) using `FT_LOGE` and the rest (raw info) using `FT_LOGE_DBG`? If so, it's not a good option.

Reply:
Thanks for detailed explanation :) Indeed, it may be better to just use `FT_LOG{E, W, I, D}` instead of introducing new functions.